FABulous/FABulous.py

```diff
@@ -125,23 +125,33 @@ def main():
 
     setup_global_env_vars(args)
 
-    args.top = os.getenv("FAB_PROJ_DIR").split("/")[-1]
+    if p := os.getenv("FAB_PROJ_DIR"):
+        projectDir = Path(p).absolute()
+    else:
+        projectDir = Path(args.project_dir).absolute()
+
+    args.top = projectDir.stem
 
     if args.createProject:
-        create_project(os.getenv("FAB_PROJ_DIR"), args.writer)
+        create_project(projectDir, args.writer)
         exit(0)
 
-    if not os.path.exists(f"{os.getenv('FAB_PROJ_DIR')}/.FABulous"):
+    if not projectDir.exists():
+        logger.error(f"The directory provided does not exist: {projectDir}")
+        exit(1)
+
+    if not (projectDir / ".FABulous").exists():
         logger.error(
             "The directory provided is not a FABulous project as it does not have a .FABulous folder"
         )
-        exit(-1)
+        exit(1)
```

Comment:
Also same here, if it's an error, we should throw one.

Reply:
I think we should use `exit(1)` for error since it is UNIX convention.

Reply:
Sorry, you are right, I've mixed it up.